docs/DevGuide/Parallelization.md

```diff
@@ -564,6 +571,78 @@ calls on all array elements occurred in the same order. It is **undefined**
 behavior if the contribute calls are made in different orders on different array
 elements.
 
+# Mutable items in the GlobalCache
+
+Most items in the GlobalCache are constant, and are determined at
+initialization by type aliases called `const_global_cache_tags` as
+described above. However, the GlobalCache can also store mutable
+items. Because of asynchronous execution, care must be taken when mutating
+items in the GlobalCache, as described below.
+
+## 1. Initialization of mutable GlobalCache items
+
+Mutable GlobalCache items are initialized by specifying them via a
+type alias `mutable_global_cache_tags`, which is treated the same way
+as `const_global_cache_tags` for const items.
+
+## 2. Use of mutable GlobalCache items
+
+### 1. Checking if the item is up to date
+
+Because execution is asynchronous, any code that uses a mutable item
+in the GlobalCache must first check whether that item is up to date.
+The information about whether an item is up to date is assumed to be
+stored in the item itself.  For example, a mutable object stored in the
+GlobalCache might have type `std::map<temporal_id,T>` (for some type
+`T`), and then any code that uses the stored object can check whether
+an entry exists for a particular `temporal_id`.
+
+To check an item, use the function
+`Parallel::mutable_cache_item_is_ready`, which returns a bool
+indicating whether the item is up to date.  If the item is up to date,
+then it can be used.  `Parallel::mutable_cache_item_is_ready` takes a
+lambda as an argument.  This lambda is passed a single argument: a const
+reference to the item being retrieved.  The lambda should determine
+whether the item is up to date. If so, it should return boost::none;
+if not, it should return a boost::optional<CkCallback> to a callback
+function that will be called once the item becomes up to date.
+
+For the case of iterable actions, `Parallel::mutable_cache_item_is_ready`
+is typically called from the `is_ready` function of the iterable action,
+and the callback is `perform_algorithm()`.  In the example below, the
+vector is considered up to date if it is non-empty:
+
+\snippet Test_AlgorithmGlobalCache.cpp check_mutable_cache_item_is_ready
+
+Note that `Parallel::mutable_cache_item_is_ready` is called on a local
+core and does no parallel communication.
+
+### 2. Retrieving the item
+
+The item is retrieved using `Parallel::get` just like for constant items.
+For example, to retrieve the item `Tags::VectorOfDoubles`:
+\snippet Test_AlgorithmGlobalCache.cpp retrieve_mutable_cache_item
+
+Note that `Parallel::get` is called on a local core and does no
+parallel communication.
+
```

Comment:
We intentionally disable getting mutable tags from the DataBox.  Because if you can do that, then ComputeItems can use them, and then the callback mechanism would be easily bypassed.